cmd/dump/dump.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2023 The K8sGPT Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dump
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/k8sgpt-ai/k8sgpt/pkg/ai"
+	"github.com/k8sgpt-ai/k8sgpt/pkg/kubernetes"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"k8s.io/apimachinery/pkg/version"
+)
+
+type DumpOut struct {
+	AIConfiguration        ai.AIConfiguration
+	ActiveFilters          []string
+	KubenetesServerVersion *version.Info
+}
+
+var DumpCmd = &cobra.Command{
+	Use:   "dump",
+	Short: "Creates a dump for debugging",
+	Long:  ``,
+	Run: func(cmd *cobra.Command, args []string) {
+
+		// Fetch the configuration object(s)
+		// get ai configuration
+		var configAI ai.AIConfiguration
+		err := viper.UnmarshalKey("ai", &configAI)
+		if err != nil {
+			color.Red("Error: %v", err)
+			os.Exit(1)
+		}
+
+		for _, config := range configAI.Providers {
+			// blank out the password
+			config.Password = ""
+		}
+
+		activeFilters := viper.GetStringSlice("active_filters")
+		// Get Kubernetes server data
+		kubecontext := viper.GetString("kubecontext")
+		kubeconfig := viper.GetString("kubeconfig")
+		client, err := kubernetes.NewClient(kubecontext, kubeconfig)
+		if err != nil {
+			color.Red("Error: %v", err)
+			os.Exit(1)
+		}
+		version, err := client.Client.Discovery().ServerVersion()
+		if err != nil {
+			color.Yellow("Could not find kubernetes server version")
+		}
+		var dumpOut DumpOut = DumpOut{
+			AIConfiguration:        configAI,
```

Comment:
Shouldn't we obfuscate sensitive fields (such as the API key of the AI provider)? This information is quite sensitive, and I think it would be good to avoid displaying it in plain text but perhaps only show the first 4-5 characters or the last ones.

I would like to avoid a situation where this command could lead to the leakage of sensitive information. I have a potential use case in mind. If we add this command in "server" mode, a client could, for example, use this command to retrieve the API key of our AI providers and thus expose sensitive data.

What do you think?

Reply:
````
type AIProvider struct {
	Name           string        `mapstructure:"name"`
	Model          string        `mapstructure:"model"`
	Password       string        `mapstructure:"password" yaml:"password,omitempty"`
	BaseURL        string        `mapstructure:"baseurl" yaml:"baseurl,omitempty"`
	ProxyEndpoint  string        `mapstructure:"proxyEndpoint" yaml:"proxyEndpoint,omitempty"`
	ProxyPort      string        `mapstructure:"proxyPort" yaml:"proxyPort,omitempty"`
	EndpointName   string        `mapstructure:"endpointname" yaml:"endpointname,omitempty"`
	Engine         string        `mapstructure:"engine" yaml:"engine,omitempty"`
	Temperature    float32       `mapstructure:"temperature" yaml:"temperature,omitempty"`
	ProviderRegion string        `mapstructure:"providerregion" yaml:"providerregion,omitempty"`
	ProviderId     string        `mapstructure:"providerid" yaml:"providerid,omitempty"`
	CompartmentId  string        `mapstructure:"compartmentid" yaml:"compartmentid,omitempty"`
	TopP           float32       `mapstructure:"topp" yaml:"topp,omitempty"`
	TopK           int32         `mapstructure:"topk" yaml:"topk,omitempty"`
	MaxTokens      int           `mapstructure:"maxtokens" yaml:"maxtokens,omitempty"`
	OrganizationId string        `mapstructure:"organizationid" yaml:"organizationid,omitempty"`
	CustomHeaders  []http.Header `mapstructure:"customHeaders"`
}
````

I think password is the only sensitive field there? Happy to be proven wrong.

Reply:
Done!